docs/design/trimmed-assemblies.md

```diff
@@ -0,0 +1,212 @@
+
+The attribute will survive trimming like other assembly-level attributes do.
+
+If `IsTrimmable` MSBuild metadata is set for an assembly, this overrides the `IsTrimmable` attribute. This allows a developer to opt an assembly into trimming even if it does not have the attribute, or to disable trimming of an assembly that has the attribute.
+
+Instead of using `IsTrimmable` metadata in the SDK to control trimmable assemblies, we will move to a model where all trimmable SDK assemblies are built with `[assembly: AssemblyMetadata("IsTrimmable", "True")]`.
+
+### `TrimmableAssembly`
+
+This ItemGroup contains assembly names that will be opted into trimming via `IsTrimmable` metadata. For simple cases, this provides an easier way to enable trimming of additional assemblies, without requiring a custom MSBuild target.
+
+```xml
+<ItemGroup>
+  <TrimmableAssembly Include="MyAssembly" />
+</ItemGroup>
+```
+
+The above will opt `MyAssembly.dll` into trimming. Note that the ItemGroup should contain assembly names without an extension, similar to [`TrimmerRootAssembly`](https://docs.microsoft.com/en-us/dotnet/core/deploying/trimming-options#root-assemblies). Before .NET 6 this would have been done with a target:
+
+```xml
+<Target Name="ConfigureTrimming"
+        BeforeTargets="PrepareForILLink">
+  <ItemGroup>
+    <ManagedAssemblyToLink Condition="'%(Filename)' == 'MyAssembly'">
+      <IsTrimmable>true</IsTrimmable>
+    </ManagedAssemblyToLink>
+  </ItemGroup>
+</Target>
+```
+
+## Future evolution
+
+As the .NET ecosystem shifts to support trimming of more libraries, we will be able to rely more on the trim analysis warnings to provide correctness guarantees. Since these guarantees are the same regardless of the `TrimMode`, we expect SDKs to move to more aggressive trimming defaults.
+
+We expect that the .NET SDK will eventually set `TrimMode` to `link` instead of `copyused` as it does today. Long-term, we may even go as far as enabling trimming of all assemblies by default when using ILLink. Existing MSBuild `IsTrimmable` and `TrimMode` metadata will continue to make it possible for the developer to opt in or out of trimming. We could additionally provide further knobs to simplify controlling trimming behavior and opting out of such defaults.
+
+### `TrimAllAssemblies` global opt-in
+
+We could make it it easier to enabling trimming for all assemblies with a simple boolean. This would be equivalent to setting `IsTrimmable` to `true` on every assembly that is input to the linker. For example:
+
+```xml
+<PropertyGroup>
+  <TrimAllAssemblies>true</TrimAllAssemblies>
+</PropertyGroup>
+```
+
+could be used instead of
+
+```xml
+<Target Name="ConfigureTrimming"
+        BeforeTargets="PrepareForILLink">
+  <ItemGroup>
+    <ManagedAssemblyToLink>
+      <IsTrimmable>true</IsTrimmable>
+    </ManagedAssemblyToLink>
+  </ItemGroup>
```

Comment:
This actually doesn't work as it is written. You need to add 

```xml
      <!-- Pass the app assembly as a root -->
      <TrimmerRootAssembly Include="@(IntermediateAssembly)" />
```

Reply:
Thanks, I forgot about this...